browser/sync/brave_sync_client_unittest.cc

```diff
@@ -0,0 +1,147 @@
+  return profile_builder.Build();
+}
+
+void FakeAdBlockSubscriptionDownloadManagerGetter(
+    base::OnceCallback<
+        void(brave_shields::AdBlockSubscriptionDownloadManager*)>) {
+  // no-op, subscription services are not currently used in unit tests
+}
+
+}  // namespace
+
+namespace browser_sync {
+
+class BraveSyncClientTest : public testing::Test {
+ public:
+  BraveSyncClientTest() {}
+  ~BraveSyncClientTest() override {}
+
+ protected:
+  void SetUp() override {
+    EXPECT_TRUE(temp_dir_.CreateUniqueTempDir());
+    profile_ = CreateProfile(temp_dir_.GetPath());
+    EXPECT_TRUE(profile_.get() != NULL);
+
+    SetupAdblockServiceForBraveBrowserProcess();
+  }
+
+  Profile* profile() { return profile_.get(); }
+
+ private:
+  void SetupAdblockServiceForBraveBrowserProcess();
+
+  // Need this as a very first member to run tests in UI thread
+  // When this is set, class should not install any other MessageLoops, like
+  // base::test::ScopedTaskEnvironment
+  content::BrowserTaskEnvironment task_environment_;
+  std::unique_ptr<Profile> profile_;
+  base::ScopedTempDir temp_dir_;
+
+  std::unique_ptr<ScopedTestingLocalState> local_state_;
+  std::unique_ptr<TestingBraveComponentUpdaterDelegate>
+      brave_component_updater_delegate_;
+};
+
+// We need this because otherwise we'll get crash on uninitialized
+// ad_block_service_ at
+//    component_factory_->CreateCommonDataTypeControllers() =>
+//    CreateAdBlockSubscriptionDownloadClient() =>
+//    g_brave_browser_process->ad_block_service()
+void BraveSyncClientTest::SetupAdblockServiceForBraveBrowserProcess() {
+  local_state_ = std::make_unique<ScopedTestingLocalState>(
+      TestingBrowserProcess::GetGlobal());
+
+  brave_component_updater_delegate_ =
+      std::make_unique<TestingBraveComponentUpdaterDelegate>(
+          local_state_->Get());
+
+  base::FilePath user_data_dir;
+  DCHECK(base::PathService::Get(chrome::DIR_USER_DATA, &user_data_dir));
+  auto adblock_service = std::make_unique<brave_shields::AdBlockService>(
+      brave_component_updater_delegate_->local_state(),
+      brave_component_updater_delegate_->locale(), nullptr,
+      brave_component_updater_delegate_->GetTaskRunner(),
+      std::make_unique<brave_shields::AdBlockSubscriptionServiceManager>(
+          brave_component_updater_delegate_->local_state(),
+          brave_component_updater_delegate_->GetTaskRunner(),
+          base::BindOnce(&FakeAdBlockSubscriptionDownloadManagerGetter),
+          user_data_dir));
+
+  TestingBraveBrowserProcess::GetGlobal()->SetAdBlockService(
+      std::move(adblock_service));
+}
+
+TEST_F(BraveSyncClientTest, CreateDataTypeControllersSearchEngines) {
+  auto sync_client =
+      std::make_unique<browser_sync::ChromeSyncClient>(profile());
+
+  syncer::TestSyncService service;
+  const syncer::DataTypeController::TypeVector controllers =
+      sync_client->CreateDataTypeControllers(&service);
+
+  auto search_engines_controller_it = std::find_if(
```

Comment:
nit: Maybe use base::ranges::find_if?

Reply:
`ranges::any_of` even better, can be used directly as a test expectation.